src/SignalR/clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -352,12 +352,20 @@ export class HttpConnection implements IConnection {
         const transportExceptions: any[] = [];
         const transports = negotiateResponse.availableTransports || [];
         for (const endpoint of transports) {
+            const transportOrError = this.resolveTransportOrError(endpoint, requestedTransport, requestedTransferFormat);
+            if (transportOrError instanceof Error) {
+                // Store the error and continue, we don't want to cause a re-negotiate in these cases
+                transportExceptions.push(`${endpoint.transport} failed: ${transportOrError}`);
+            }
             try {
-                const transport = this.resolveTransport(endpoint, requestedTransport, requestedTransferFormat);
-                if (typeof transport === "number") {
-                    this.transport = this.constructTransport(transport);
+                if (typeof transportOrError === "number") {
+                    this.transport = this.constructTransport(transportOrError);
                     if (!negotiateResponse.connectionId) {
-                        negotiateResponse = await this.getNegotiationResponse(url);
+                        try {
+                            negotiateResponse = await this.getNegotiationResponse(url);
+                        } catch (ex) {
+                            return Promise.reject(ex);
+                        }
                         connectUrl = this.createConnectUrl(url, negotiateResponse.connectionId);
                     }
                     await this.transport!.connect(connectUrl, requestedTransferFormat);
```

Comment:
Can anything other than this line throw? If not, I'd like to keep the try/catch local to the transport.connect call.

Reply:
`constructTransport` can throw if we can't successfully create the requested transport.

I can move the try/catch inside the if check though

Reply:
I think that makes sense. We probably also don't need to null out negotiateResponse.connectionId and renegotiate when constructTransport throws.